core/search_index.py

```
@@ -0,0 +1,645 @@
+                            if os.path.isdir(os.path.join(base_dir, d))
+                        ]
+                    except OSError:
+                        continue
+
+                    # Purge rows for customers that no longer exist on disk.
+                    if not _cancelled():
+                        customer_set = set(customers)
+                        if customer_set:
+                            placeholders = ','.join('?' * len(customer_set))
+                            conn.execute(
+                                f"DELETE FROM jobs WHERE prefix=? AND customer NOT IN ({placeholders})",
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)